### ISSUE TITLE
Striking artifact on 3d segmentation

### ISSUE THREAD

**Issue by dobrik4ever** [MouseLand/cellpose#503]
Greetings!

I am trying to use cellpose V2.0.4 on a multiphoton z-stack (Z x W x H dimension) can be found here [dropbox](https://www.dropbox.com/s/3rfrgvxidr78hot/2c.tif?dl=0) When I do 2D segmentation it works perfectly fine. But when I am trying to do 3d, I run into a problem with the striking artifacts. 
![image](https://user-images.githubusercontent.com/44587102/164782475-f5d68758-eb9a-4899-974e-a31926c164e9.png)
I am trying to do that in Google colab on GPU:

```python
!pip install cellpose
!pip uninstall opencv-python -y
!pip install opencv-python==4.1.2.30

import numpy as np
from matplotlib import pyplot as plt
from skimage import io as sio
from cellpose import io, models

stack = sio.imread('2c.tif')[:,:,:,0]
print(stack.shape)
sio.imshow(stack[80])
plt.xticks([]); plt.yticks([]);

model = models.Cellpose(gpu=True, model_type='cyto')
channels = [0,0]
masks, flows, styles, diams = model.eval(stack, channels=channels, do_3D=True ,anisotropy=0.7736944) # This anisotropy was selected due to z sampling of 1 micron and xy is 0.7...

plt.imshow(flows[0][80])
plt.colorbar()
```

**Comment by kevinjohncutler** [comment]
@dobrik4ever This is to be expected,as the flow components are computed along each of the three axes as a set of independent flow slices before being merged (x from xy and xz planes etc.). This independent slice prediction leads to potential inconsistencies from slice to slice, hence the striping. A subtler issue is a sort of squeezing of the flow into the cardinal directions, which you can see as patchy/blocky colors in the phase diagram. 

I have generalized Omnipose to predict 'true' 3D flow fields directly from 3D volumes. The big caveat is that it has to be trained on true 3D labels. If you have such a dateset or are interested in making one, let me know. The big advantage of Cellpose's 3D implementation is that it leverages 2D segmentation to do 3D, but as you can see, that has its own quirks and limitations.

**Comment by dobrik4ever** [comment]
@kevinjohncutler Thank you a lot for your reply! I don't have any 3D dataset and would probably put my hands on trying to merge a set of 2D masks instead of trying to achieve true 3D segmentation. Anyway thanks, and good luck with this project it is a masterpiece!
